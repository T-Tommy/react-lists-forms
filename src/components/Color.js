import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class color extends PureComponent {
  static propTypes = {
    color: PropTypes.object.isRequired
  }
  render() {
    const { name, hex, rgb: { r, g, b } } = this.props.color;
    return (
      <dl>
        <dt>Name</dt>
        <dd>{ name }</dd>

        <dt>Hex</dt>
        <dd>{ hex }</dd>

        <dt>RGB</dt>
        <dd>Red - { r }</dd>
        <dd>Green - { g }</dd>
        <dd>Blue - { b }</dd>
      </dl>
    );
  }
}
