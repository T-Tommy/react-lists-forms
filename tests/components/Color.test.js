import React from 'react';
import { shallow } from 'enzyme';
import Color from '../../src/components/Color';

describe('Color component', () => {
  const color = {
    name: 'Red',
    hex: 'FF0000',
    rgb: {
      r: 255,
      g: 0,
      b: 0
    }
  };

  it('renders Color', () => {
    const wrapper = shallow(<Color color={ color } />);
    expect(wrapper).toMatchSnapshot();
  });
});
