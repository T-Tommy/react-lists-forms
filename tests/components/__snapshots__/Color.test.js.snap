// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders Color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Red
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    FF0000
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    Red - 
    255
  </dd>
  <dd>
    Green - 
    0
  </dd>
  <dd>
    Blue - 
    0
  </dd>
</dl>
`;
